spring-vault-core/src/test/java/org/springframework/vault/core/VaultVersionedKeyValueTemplateIntegrationTests.java

```diff
@@ -103,6 +104,34 @@ void shouldCreateVersionedWithCAS() {
 			.hasMessageContaining("check-and-set parameter did not match the current version");
 	}
 
+
+	@Test
+	void shouldWriteSecretWithCustomMetadata(){
+		Person person = new Person();
+		person.setFirstname("Walter");
+		person.setLastname("White");
+
+		String key = UUID.randomUUID().toString();
+
+		Map<String, String> customMetadata = new HashMap<>();
+		customMetadata.put("foo", "bar");
+		customMetadata.put("uid", "werwer");
+
+		Metadata metadata= Metadata.builder()
+				.customMetadata(customMetadata)
+				.createdAt(Instant.now())
+				.version(Version.unversioned())
+				.build();
+
+		Metadata metadata1 = this.versionedOperations.put(key, Versioned.create(person, metadata));
```

Comment:
Apologies for asking good sir @mp911de, but this test is failing as the customMetadata seems to be lost. Do you know where I am missing a step?